WordPress/src/main/java/org/wordpress/android/ui/uploads/AbstractUploadHandler.java

```diff
@@ -0,0 +1,7 @@
+package org.wordpress.android.ui.uploads;
+
+abstract class AbstractUploadHandler {
+    abstract void unregister();
+    abstract boolean hasInProgressUploads();
+    abstract void cancelInProgressUploads();
+}
```

Comment:
Two thoughts here:

1) if we keep this, can we add an `abstract void register(Object object)` method to this interface? I think it'd be balanced with `unregister()` this way
Then we can have this abstract class have a default constructor that calls `register()`, and the subclasses would call `super()` in their own constructors as well.

2) Jumping on another point of view: did you consider making this an interface? maybe removing `unregister`  and making that a part / implicit responsibility of each `UploadHandler` implementation, and leaving clear that `hasInProgressUploads()` and `cancelInProgressUploads()` are the actual capabilities any class implementing the interface should have.

Thoughts?